test/LibraryManager.IntegrationTest/Services/InstallDialogTestExtension.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Microsoft.Test.Apex.Services;
+using Microsoft.Test.Apex.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.Web.LibraryManager.Vsix.UI;
+
+namespace Microsoft.Web.LibraryManager.IntegrationTest.Services
+{
+    public class InstallDialogTestExtension : VisualStudioInProcessTestExtension<object, InstallDialogVerifier>
+    {
+        /// <summary>
+        /// Add client side libraries dialog test extension for interaction with visual studio inprocess types
+        /// </summary>
+        public IInstallDialogTestContract InstallDialog
+        {
+            get
+            {
+                return this.ObjectUnderTest as IInstallDialogTestContract;
+            }
+        }
+
+        public void SetLibrary(string library)
+        {
+            UIInvoke(() =>
+            {
+                this.InstallDialog.Library = library;
+            });
+        }
+
+        public void WaitForFileSelections()
+        {
+            WaitFor.IsTrue(() =>
+            {
+                return UIInvoke(() =>
+                {
+                    return this.InstallDialog.IsAnyFileSelected;
+                });
+            }, TimeSpan.FromSeconds(20), conditionDescription: "File list not loaded");
+        }
+
+        public void ClickInstall()
```

Comment:
>void [](start = 15, length = 4)

Would it make sense for this to be async such that the caller could await it and not have to figure out when things are done?

Reply:
This is called from test. I think its cleaner this way. We could do below:

```
public async Task ClickInstallAsync()
{
    await UIInvoke(async() =>
    { 
        Task t = InstallDialog.ClickInstallAsync();
        await t;
    });
}
```
and use JTF in test to await on the task. But I would like to do less in the test. Let me know your thoughts.

---
In reply to: [205451840](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205451840) [](ancestors = 205451840)

Reply:
But the test needs to wait for it to be completed one way or the other, right?

---
In reply to: [205864705](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205864705) [](ancestors = 205864705,205451840)

Reply:
Leaving this as is. As discussed offline, the dialog is dismissed when install button is clicked. So we don't have to wait on anything here.

---
In reply to: [205905301](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205905301) [](ancestors = 205905301,205864705,205451840)